src/parser/ast/YieldExpressionNode.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+#ifndef YieldExpressionNode_h
+#define YieldExpressionNode_h
+
+#include "ExpressionNode.h"
+
+namespace Escargot {
+
+class YieldExpressionNode : public ExpressionNode {
+public:
+    YieldExpressionNode(RefPtr<Node> argument, bool isDelegate)
+        : ExpressionNode()
+        , m_argument(argument)
+        , m_isDelegate(isDelegate)
+    {
+    }
+
+    virtual ASTNodeType type() { return ASTNodeType::YieldExpression; }
+    virtual void generateExpressionByteCode(ByteCodeBlock* codeBlock, ByteCodeGenerateContext* context, ByteCodeRegisterIndex dstRegister)
+    {
+        if (m_argument == nullptr) {
+            codeBlock->pushCode(Yield(ByteCodeLOC(m_loc.index), SIZE_MAX, dstRegister), context, this);
```

Comment:
If `SIZE_MAX` is used to mark an invalid register, `REGISTER_LIMIT` would be suitable for this purpose.